wagtail_streamfield_migration_toolkit/management/commands/streamdatamigration.py

```diff
@@ -0,0 +1,230 @@
+
+        for path in self.paths:
+            model_name, field_name, block_path = self.parse_path(path)
+            operation_and_block_path = self.operation_maker(
+                block_path=block_path,
+                model_name=model_name,
+                field_name=field_name,
+                **options
+            )
+            key = (model_name, field_name)
+            if key in operations_and_block_paths_by_model_field:
+                operations_and_block_paths_by_model_field[key].append(
+                    operation_and_block_path
+                )
+            else:
+                operations_and_block_paths_by_model_field[key] = [
+                    operation_and_block_path
+                ]
+
+        migration = Migration(self.migration_name, self.app_label)
+        migration.operations = []
+        for (
+            model_name,
+            field_name,
+        ), operations_and_block_paths in (
+            operations_and_block_paths_by_model_field.items()
+        ):
+            migration.operations.append(
+                MigrateStreamData(
+                    app_name=self.app_label,
+                    model_name=model_name,
+                    field_name=field_name,
+                    operations_and_block_paths=operations_and_block_paths,
+                )
+            )
+
+        # If the user doesn't give a name for the migration file, generate one based on the
+        # operations used.
+        if not self.migration_name:
+            self.migration_name = "_".join(self._migration_names)[:30]
+
+        autodetector = MigrationAutodetector(
+            self.project_state, ProjectState.from_apps(apps)
+        )
+        changes = {self.app_label: [migration]}
+        changes = autodetector.arrange_for_graph(
+            changes=changes,
+            graph=loader.graph,
+            migration_name=self.migration_name,
+        )
+
+        # we're going to have only one value here though
+        for app_label, app_migrations in changes.items():
+            for migration in app_migrations:
+                writer = MigrationWriter(migration, True)
+                migration_string = writer.as_string()
+                with open(writer.path, "w", encoding="utf-8") as fh:
+                    fh.write(migration_string)
+
+    def make_rename_operation(
+        self, block_path, model_name, field_name, old_name, new_name, **options
+    ):
+
+        block_def = self.get_block_def(
+            model_name=model_name, field_name=field_name, block_path=block_path
+        )
+
+        if isinstance(block_def, StreamBlock):
+            data_operation = RenameStreamChildrenOperation
+        elif isinstance(block_def, StructBlock):
+            data_operation = RenameStructChildrenOperation
+        else:
+            raise CommandError(
+                "Invalid block structure {} for rename operation.".format(block_def)
+            )
+
+        self._migration_names.append(
```

Comment:
What if you have the same rename for multiple paths? You might end up with repeated names in the title. Why not use a set here, or similar, so you only get distinct names?